### ISSUE TITLE
Update README.md

### ISSUE THREAD

**Issue by destiny-yy** [Anonymous-222/EmbeddingCompression#1]
# Hetu Embedding Memory Compression Experimental Analysis
Implementation of the Paper: "Experimental Analysis of High-dimensional Learnable Vector
Storage Compression" in Hetu

### Instructions

1.We have provided our codes in `\EmbeddingCompression\examples\ctr` directory.You can check the model architecture of models or add new models in the `models` directory.  
2.All the implementation of embedding compression methods mentioned in the paper can be found in`\EmbeddingCompression\python\hetu\layers\compressed_embedding.py`.You can also add new methods in the above file.  
3.All the trainer code are placed in `\EmbeddingCompression\python\hetu\scheduler`.If you want to use new embedding methods,you should not only implement the methods codes but the trainer codes and put it in the above directory.  
4.All of the gpu ops are implemented and placed in `\EmbeddingCompression\python\hetu\gpu_ops`.  
5.You can find the implementations of optimizer in `\EmbeddingCompression\python\hetu\optimizer.py`,if you want to use new optimizer, please add your codes in the above file.  
### Training
```
usage: run_compressed.py [--model MODEL] [--method METHOD] [--phase PHASE]
               [--ectx ECTX] [--ctx CTX] [--bs BS] [--opt OPT] [--dim DIM] [--lr LR]
               [--dataset DATASET] [--data_path DATA_PATH] [--nepoch NEPOCH] 
               [--num_test_every_epoch NUM_TEST_EVERY_EPOCH] [--seed SEED] 
               [--separate_fields --SEPERATE_FIELDS] [--use_multi USE_MULTI]
               [--compress_rate COMPRESS_RATE] [--logger LOGGER] [--run_id RUN_ID]
               [--debug DEBUG] [--threshold THRESHOLD] [--stage STAGE] [--load_ckpt LOAD_CKPTS]
               [--log_dir LOG_DIR] [--save_dir SAVE_DIR] [--save_topk SAVE_TOPK] 
               [--check_val CHECK_VAL] [--check_test CHECK_TEST] 
               [--early_stop_steps EARLY_STOP_STEPS]
               
optional arguments:
  --model               Model to be tested
  --method              Method to be used
  --phase               Train or Test
  --ectx                Context for embedding table
  --ctx                 Context for model
  --bs                  Batch size to be used
  --opt                 Optimizer to be used
  --dim                 Dimension to be used
  --lr                  Learning rate to be used
  --dataset             Dataset to be used 
  --data_path           Path to dataset
  --nepoch              Num of epochs 
  --num_test_every_epoch
                        Evaluate each 1/100 epoch in default(default:100)
  --seed                Random seed
  --separate_fields     Whether seperate fields
  --use_multi           Whether use multi embedding
  --compress_rate       Compress rate to be used
  --logger              Logger to be used
  --run_id              Run id to be logged
  --debug               Whether in debug mode
  --threshold           The threshold for inter-feature compression
  --stage               The start stage for train/test
  --load_ckpt           CheckPoints to be used
  --log_dir             Directory for logging
  --save_dir            Directory for saving
  --save_topk           Number of ckpts to be saved
  --check_val           Whether check validation data during training
  --check_test          Whether check test data during training
  --early_stop_steps    Early stopping if no improvement over steps
```
An example of training usage is shown as follows:
```
python run_compressed.py --model dlrm --method compo --compress_rate 0.5 --ctx 0 --nepoch 1000 --eraly_stop 40 --opt adam --dim 16 --lr 0.001 
```

### Evaluation
```
usage: run_compressed.py [-model MODEL] [--ctx CTX] [--method METHOD] [--dataset DATASET]
               [--opt OPT] [--load_ckpt CHECKPOINTS] [--compress_rate CR] [--phase test]

```
An example of training usage is shown as follows:
```
python run_compressed.py --model dlrm --method compo --compress_rate 0.5 --ctx 0 --phase test
--load_ckpt r'check points path'
```

### Prepare Training dataset
  - Avazu is provided by Avazu, a mobile advertising platform, at the 2015 ctr prediction competition on Kaggle. It contains 10 days of real data, 40M records in total. There are 22 feature fields for a total of 9.4 million unique features.Avazu can be downloaded manually from https://www.kaggle.com/c/avazu-ctr-prediction/data.

  - Criteo is part of the 7-day real data provided by the Criteo Lab on Kaggle in 2014. with a total of 46M records. It contains 13 continuous features and 26 feature field with a total of 34M unique features. Criteo is also adopted in MLPerf, a standard benchmark for machine learning performance.Criteo can be downloaded manually from 
  https://s3-eu-west-1.amazonaws.com/kaggle-display-advertising-challenge-dataset/dac.tar.gz.

  - Details of the two datasets

| Datasets        | Samples          | Features    | Fields |
| ------------- |:-------------:|:-----:| -------:|
| Avazu      | 40428967      | 9449445   | 26 |
| Criteo     | 45840617      | 33762577  | 2  |

- You can create a new folder named datasets under the path `\EmbeddingCompression\examples\ctr`.
  We also provide scripts `load_data.py`,`load_data_avazu_variants.py` and `load_data_criteo_variants.py` for processing datasets under the path `\EmbeddingCompression\examples\ctr\models`.

**Comment by HugoZHL** [comment]
Merged
